### PR TITLE
Add client function to create Resource from raw json

### PR DESCRIPTION
**Pull request by tholam**
Implementing the feature request I made here 
https://github.com/pycontribs/jira/issues/743

### CODE REVIEW

**Comment by tholam** [comment]
Hmm not sure why the linting failed on stuff I didn't change and I'm not too familiar with flake8.  Will take a look at it.  Also test for attachment failed because the uploadLimit in the attachment response was different from the expected value.  Can someone more familiar with this help me out with this test failure?

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 argparse; python_version<'2.7'
 defusedxml
 functools32; python_version<'3.2'  # PSF license
-oauthlib[signedtoken]>=1.0.0  # dep of requests-oauthlib, to avoid pycrypto. See: https://github.com/pycontribs/jira/issues/619
+oauthlib[signedtoken]>=1.0.0,<3.0.0  # dep of requests-oauthlib, to avoid pycrypto. See: https://github.com/pycontribs/jira/issues/619
```
[the remark this answers is not in the document]
Comment (Fruglemonkey) [reply requirements.txt]: We should just remove the oauthlibs requirement entirely. Was about to comment here https://github.com/pycontribs/jira/pull/627 before I saw this PR.

`requests-oauthlibs>=0.6.1` from the current requirements will pull in the latest `requests-oauthlibs` anyway, which now defines a similar set of oatuhlib requirements: https://github.com/requests/requests-oauthlib/blob/v1.1.0/requirements.txt
`oauthlib[signedtoken]>=2.1.0,<3.0.0`

pip's dependency handling is causing us to hit the issue that this commit addresses: https://github.com/pypa/pip/issues/988#issuecomment-94205646

[the remark this answers is not in the document]
Comment (tholam) [reply requirements.txt]: Should I remove it from here as well? https://github.com/pycontribs/jira/blob/master/requirements-dev.txt#L8

[the remark this answers is not in the document]
Comment (brainwane) [reply requirements.txt:4]: Regarding the pip dependency handling: anyone interested in this PR, please [try the beta of the new resolver in pip 20.2](https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-2-2020) and see whether that helps?

**Pull request closed without merging**